DataStructure/src/main/list/ResizableArray.java

```diff
@@ -0,0 +1,167 @@
+package main.list;
+
+/**
+ * Represents a resizable array of integer elements.
+ */
+
+public class ResizableArray {
+    private int size;
+    private int firstBlankIndex;
+    private int list[];
+
+    /**
+     * Constructs a resizable array
+     * @param size
+     * size of the list
+     */
+    public ResizableArray(int size){
+        this.size = size;
+        list = new int[this.size] ;
+        firstBlankIndex = 0;
+    }
+
+    /**
+     * Constructs a resizable array with default size 50
+     */
+    public ResizableArray(){
+        size = 50;
+        list = new int[50] ;
+        firstBlankIndex = 0;
+    }
+
+    /**
+     * Adds an element to the list
+     * @param num
+     */
+    public void add(int num){
+        if (size == 0){
+            list[size] = num;
+            return;
+        }
+        else if (firstBlankIndex == size){
+            size = size*2 ;
+            int list2 [] = new int[size];
+            for (int i = 0; i < firstBlankIndex; i++) {
+               list2[i] = list[i];
+            }
+            list = list2;
+        }
```

Comment:
1. Move the content of else if block to a separate method (name - ensureCapacity)
2. Use arrays copyOf method instead of manual copy for creating a larger array.
3. Call the ensureCapacity method as first line in this method
4. After ensuring capacity, add the element at the first blank index.